mitiq/shadows/test/test_quantum_processing.py

```diff
@@ -0,0 +1,230 @@
+        f"{(n_total_measurements, n_qubits)}, got {shadow_outcomes.shape}"
+    )
+    assert pauli_strings.shape == (n_total_measurements,), (
+        f"Pauli strings have incorrect shape, expected "
+        f"{(n_total_measurements, n_qubits)}, got {pauli_strings.shape}"
+    )
+    assert len(pauli_strings[0]) == n_qubits, (
+        f"Pauli strings have incorrect number of characters, "
+        f"expected {n_qubits}, got {len(pauli_strings[0])}"
+    )
+
+
+@pytest.mark.parametrize("n_qubits", [1, 2, 5])
+@pytest.mark.parametrize("executor", [cirq_executor, qiskit_executor])
+def test_random_pauli_measurement_output_types(
+    n_qubits: int, executor: Callable
+):
+    """Test that random_pauli_measurement returns the correct output types."""
+    qubits = cirq.LineQubit.range(n_qubits)
+    circuit = simple_test_circuit(qubits)
+    shadow_outcomes, pauli_strings = random_pauli_measurement(
+        circuit, n_total_measurements=10, executor=executor
+    )
+    assert shadow_outcomes[0].dtype == int, (
+        f"Shadow outcomes have incorrect dtype, expected int, "
+        f"got {shadow_outcomes.dtype}"
+    )
+    assert isinstance(pauli_strings[0], str), (
+        f"Pauli strings have incorrect dtype, expected str, "
+        f"got {pauli_strings.dtype}"
+    )
+
+
+@pytest.mark.parametrize("executor", [cirq_executor, qiskit_executor])
+def test_random_pauli_measurement_time_growth(executor: Callable):
+    """Test that random_pauli_measurement scales linearly with the
+    number of measurements."""
+    n_qubits = 5
+    qubits = cirq.LineQubit.range(n_qubits)
+    circuit = simple_test_circuit(qubits)
+    times = []
+    measurements = [10, 20, 30, 40, 50]
+    for n in measurements:
+        start_time = time.time()
+        random_pauli_measurement(circuit, n, executor=executor)
+        times.append(time.time() - start_time)
+    for i in range(1, len(times)):
+        assert times[i] / times[i - 1] == pytest.approx(
+            measurements[i] / measurements[i - 1], rel=8
+        )
+
+
+@pytest.mark.parametrize("executor", [cirq_executor, qiskit_executor])
+def test_random_pauli_measurement_time_power_growth(
+    executor: Callable,
+):
+    """Test that random_pauli_measurement scales follow power law with the
+    number of measurements."""
+    n_qubits = [3, 6, 9, 12, 15]
+
+    times = []
+    for n in n_qubits:
+        qubits = cirq.LineQubit.range(n)
+        circuit = simple_test_circuit(qubits)
+        start_time = time.time()
+        random_pauli_measurement(
+            circuit,
+            100,  # number of total measurements
+            executor=executor,
+        )
+
+        times.append(time.time() - start_time)
+    for i in range(1, len(times)):
+        log_ratio_times = np.log(times[i] / times[i - 1])
+        log_ratio_qubits = np.log(n_qubits[i] / n_qubits[i - 1])
+        assert log_ratio_times == pytest.approx(log_ratio_qubits, rel=5)
```

Comment:
Could we remove this test? It's quite slow, and it's not testing the correctness of the code. I ran `pytest mitiq/shadows --durations=15` to get the tests ordered by speed in descending order.
```
==================================================================== slowest 15 durations =============================================
13.96s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_time_power_growth[qiskit_executor]
3.96s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_time_growth[qiskit_executor]
1.14s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_time_power_growth[cirq_executor]
0.40s call     mitiq/shadows/test/test_quantum_processing.py::test_generate_random_pauli_strings_time
0.30s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_output_types[qiskit_executor-2]
0.30s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_output_dimensions[qiskit_executor-5]
0.30s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_output_dimensions[qiskit_executor-2]
0.29s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_no_errors[qiskit_executor-5]
0.29s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_output_types[qiskit_executor-5]
0.28s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_no_errors[qiskit_executor-1]
0.27s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_no_errors[qiskit_executor-2]
0.20s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_output_types[qiskit_executor-1]
0.20s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_output_dimensions[qiskit_executor-1]
0.15s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_time_growth[cirq_executor]
0.01s call     mitiq/shadows/test/test_quantum_processing.py::test_random_pauli_measurement_output_dimensions[cirq_executor-5]
```